File: backend/app/schemas/geo_context.py
"""Schemas for geo.admin context overlay responses."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class GeoLayerResult(BaseModel):
    """A single geo overlay layer result."""

    source: str
    label: str
    raw_attributes: dict[str, Any] = {}
    # Layer-specific parsed fields (optional, depend on layer type)
    zone: str | None = None
    value: str | None = None
    level_db: float | str | None = None
    suitability: str | None = None
    potential_kwh: float | str | None = None
    hazard_level: str | None = None
    zone_type: str | None = None
    status: str | None = None
    category: str | None = None
    name: str | None = None
    quality_class: str | None = None
    network_name: str | None = None


class GeoContextResponse(BaseModel):
    """Full geo context response for a building."""

    context: dict[str, Any] = {}
    fetched_at: str | None = None
    source_version: str | None = None
    cached: bool = False
    error: str | None = None
    detail: str | None = None


class GeoContextRefreshResponse(BaseModel):
    """Response after a forced refresh."""

    context: dict[str, Any] = {}
    fetched_at: datetime | None = None
    source_version: str = "geo.admin-v1"
    layers_count: int = 0

File: frontend/src/api/geoContext.ts
import { apiClient } from '@/api/client';

export interface GeoLayerResult {
  source: string;
  label: string;
  raw_attributes: Record<string, unknown>;
  zone?: string | null;
  value?: string | null;
  level_db?: number | string | null;
  suitability?: string | null;
  potential_kwh?: number | string | null;
  hazard_level?: string | null;
  zone_type?: string | null;
  status?: string | null;
  category?: string | null;
  name?: string | null;
  quality_class?: string | null;
  network_name?: string | null;
}

export interface GeoContextResponse {
  context: Record<string, GeoLayerResult>;
  fetched_at: string | null;
  source_version: string | null;
  cached: boolean;
  error?: string | null;
  detail?: string | null;
}

export const geoContextApi = {
  get: async (buildingId: string): Promise<GeoContextResponse> => {
    const response = await apiClient.get<GeoContextResponse>(`/buildings/${buildingId}/geo-context`);
    return response.data;
  },

  refresh: async (buildingId: string): Promise<GeoContextResponse> => {
    const response = await apiClient.post<GeoContextResponse>(`/buildings/${buildingId}/geo-context/refresh`);
    return response.data;
  },
};
